AskQ/Controllers/UserController.cs

```diff
@@ -32,27 +32,35 @@ public async Task<IActionResult> UserProfileAsync(string username, int page = 1)
                 return NotFound();
             }
 
-            IEnumerable<QuestionViewModel> questions = _dbContext.Questions.Include(q => q.Replies)
-                    .Where(q => q.AskedToGuid == user.Id && q.Replies.Any())
-                    .Skip(10 * (page - 1))
-                    .Take(10)
-                    .OrderByDescending(q => q.DateTime)
-                    .Select(q => new QuestionViewModel
-                    {
-                        Id = q.Id,
-                        QuestionText = q.QuestionText,
-                        AskedFromUsername = (_userManager.FindByIdAsync(q.AskedFromGuid)).Result.UserName,
-                        DateTime = q.DateTime,
-                        Replies = q.Replies.Select(r => r.ReplyText)
-                    }).AsEnumerable(); // Looks too long.
-                                       // There could be a better approach to populate IEnumerable<QuestionViewModel> from the IEnumerable<Question>;
+
+            // We'll keep it for now like this, but there are issues generally.
+            var questions = await _dbContext.Questions
+                                            .Include(q => q.Replies)
+                                            .Where(q => q.AskedToGuid == user.Id && q.Replies.Any()) // This query will be processed in memory. All records will be retrieved from DB.
```

Comment:
> // This query will be processed in memory. All records will be retrieved from DB.

@fiseni, Do you mean this can be problematic if the database is huge? How this can be improved?

Reply:
Efcore still doesn't support filtered includes, simply will get all the questions for that user, and then filter it only the ones with responses. Btw, why do you need that condition? Simply show the questions without responses as well.

Reply:
@fiseni

> Efcore still doesn't support filtered includes

That means I can't have "Include" after "Where" right?

> why do you need that condition?

I made the questions without responses to be available only to the user who is asked that question, and the question becomes public only after it's answered (that's the way the real ask.fm behaves).

Reply:
No, u can have, the where will be for the question. What i mean with include filter is smth like .Include(x=>x.replies.where(y=>something)) or where(x=>x.replies.where(something)). The conditions for the replies will be done in memory.

Reply:
Got it! Thanks a lot :heart:

Reply:
I you want merge everything to master or any branch that has most updated code, and I will do few changes. From tomorrow I won't be very available, so I can help you for the next few hours.

Reply:
@fiseni The current master has the most updated code. I'm really glad with your help :)

Reply:
Ok great, I'll work on it.
In the meantime explain what you want to achieve in short lines. You have profile page and question page...

Reply:
@fiseni Thanks. Here is what the site is about:
Users on the site can be asked questions (anonymously (from non-logged in user) or from another logged in user).

- The profile page of each user should show the **already** answered questions.
- The received page should show the non-answered questions with a button that allows the user to answer.
- After a question is answered, other users (logged in or anonymous) can add more replies, That's why the question Model have IEnumerable of Reply. The first reply can only be from the "AskedTo" user, because a question with no replies can only be viewed by that user.
- Currently, I didn't implement the feature where other users can add more replies. I wanted to focus on getting the current simple version on a good code state with good code quality.

Reply:
Ok, you can follow the commit to my fork for now, in the end we'll make PR.
https://github.com/fiseni/AskQ.git

Whenever you see commits starting with WIP, it means "working in progress", the solution most probably will compile with errors.

Reply:
@fiseni Great. Really thanks :)